lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -329,28 +331,29 @@ data ConsensusVersion = BFT | GenesisPraos
     deriving (Eq, Show)
 
 getConsensusVersion :: Get ConsensusVersion
-getConsensusVersion = getWord16be >>= \case
+getConsensusVersion = label "getConsensusVersion" $ getWord16be >>= \case
     1 -> return BFT
     2 -> return GenesisPraos
     other -> fail $ "Unknown consensus version: " ++ show other
 
 getNetwork :: Get Network
-getNetwork = getWord8 >>= \case
+getNetwork = label "getNetwork" $ getWord8 >>= \case
     1 -> return Mainnet
     2 -> return Testnet
     other -> fail $ "Invalid network/discrimination value: " ++ show other
 
 getMilli :: Get Milli
-getMilli = Milli <$> getWord64be
+getMilli = label "getMilli" $ Milli <$> getWord64be
 
 getLeaderId :: Get LeaderId
-getLeaderId = LeaderId <$> getByteString 32
+getLeaderId = label "getLeaderId" $ LeaderId <$> getByteString 32
 
 getLinearFee :: Get LinearFee
-getLinearFee = LinearFee
-    <$> (Quantity <$> getWord64be)
-    <*> (Quantity <$> getWord64be)
-    <*> (Quantity <$> getWord64be)
+getLinearFee = label "getLinearFee" $ do
+    const' <- Quantity <$> getWord64be
+    perByte <- Quantity <$> getWord64be
+    perCert <- Quantity <$> getWord64be
+    return $ LinearFee const' perByte perCert
```

Comment:
Reverted to do-notation for consistency with the rest of the module, to show variable names, and separate order of the binary format with the order of the `LinearFee` arguments.

Reply:
> separate order of the binary format with the order of the `LinearFee` arguments.

Fair enough!